Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md

```diff
@@ -0,0 +1,94 @@
+# Upgrading python interpreter version from 3.7.10 to 3.8.12
+
+Oppia has updated it's python interpreter version 3.7.10 in a phase of switching to a more updated interpreter version 3.8.12. For effective and effortless switching to the newer version, this guide will provide a brief and detailed instruction on how to achieve just that. Please follow the below mentioned instructions carefully.
+
+## Prerequisite
+- This page is dedicated to people who want to upgrade their python interpreter in the existing Oppia directory instead of maintaing a new local repo copy of oppia.
+- Please make sure your local repo as well as remote fork is up-to-date with the develop. 
+
+## Step 1. Installing python 3.8.12 through pyenv
+Our first step is to install the new interpreter version `3.8.12` through pyenv by issuing the following command through a terminal :
+```
+pyenv install 3.8.12
+```
+After you execute the above command you can expect some output as depicted below :
+```
+❯ pyenv install 3.8.12
+Installing Python-3.8.12...
+patching file Misc/NEWS.d/next/Build/2021-10-11-16-27-38.bpo-45405.iSfdW5.rst
+patching file configure
+patching file configure.ac
+Installed Python-3.8.12 to /home/user/.pyenv/versions/3.8.12
+```
+
+## Step 2. Determining your virtualenv manager
+
+Oppia developers and contributors use either `pyenv` or `direnv` for creating a python virtualenv and automatic switching of python version.
+
+This step will guide you through determining which one you are using currently, so that you can update your virtualenv accordingly.
+
+#### Do the following :
+
+* Go to the directory where you have installed Oppia and check if you have `.envrc` file in that directory. If the `.envrc` file is present, that means you are using direnv for virtualenv activation. `direnv` users should follow the instructions in [Step 3 (a)](#step-3-(a).-setting-up-virtualenv-for-direnv-users).
+
+* In case you don't have a `.envrc` file, check if you have `.python-version` file in that directory. If the `.python-version` file is present, that means you are using `pyenv` for virtualenv activation. `pyenv` users should follow the instructions in [Step 3 (b)](#step-3-(b).-setting-up-virtualenv-for-pyenv-users).
```

Comment:
What do they do if neither .envrc nor .python-version is present?

Reply:
Either one of them should be present . because that's how the virtualenv managers work. As we enter a directory containing these specific files, the virtualenv managers will activate the environment . And as we leave the directory the environment gets deactivated automatically. So that we don't have to manually switch the version everytime.